EXTRA_FIM/main.py

```diff
@@ -259,124 +258,11 @@ def potential_cell(self):
-        self._n_gk = self.wfile["nGk"][:]
-        off = 0
-        for ngk in self._n_gk:
-            self._fft_idx.append(self.wfile["fftIdx"][off : off + ngk])
-            off += ngk
-
-        self.k_vec = self.wfile["kVec"][:]
-
-    @property
-    def eps(self):
-        """All eigenvalues (in Hartree) as (nk,n_states) block"""
-        if self._eps is None:
-            self._check_loaded()
-            self._eps = self.wfile["eps"][:].reshape(-1, self.n_spin, self.n_states)
-        return self._eps
-
-    # Define as separate method and speed it up with numba
-    @staticmethod
-    @numba.jit
-    def _fillin(res, psire, psiim, fft_idx):
-        """Distributes condensed psi (real, imag) on full FFT mesh"""
-        rflat = res.flat
-        for ig in range(fft_idx.shape[0]):
-            rflat[fft_idx[ig]] = complex(psire[ig], psiim[ig])
-
-    def get_psi_rec(self, i, ispin, ik):
-        """Loads a single wavefunction on full FFT mesh"""
-        if i < 0 or i >= self.n_states:
-            raise IndexError(f"i={i} fails 0 <= i < n_states={self.n_states}")
-        if ispin < 0 or ispin >= self.n_spin:
-            raise IndexError(f"ispin={ispin} fails 0 <= ispin < n_spin={self.n_spin}")
-        if ik < 0 or ik >= self.nk:
-            raise IndexError(f"ik={ik} fails 0 <= ik < nk={self.nk}")
-
-        res = np.zeros(shape=self.mesh, dtype=np.complex128)
-        off = self._n_gk[ik] * (i + ispin * self.n_states)
-        psire = self.wfile[f"psi-{ik+1}.re"][off : off + self._n_gk[ik]]
-        psiim = self.wfile[f"psi-{ik+1}.im"][off : off + self._n_gk[ik]]
-        self._fillin(res, psire, psiim, self._fft_idx[ik])
-        return res
-
-    @property
-    def nk(self):
-        """Number of k-points"""
-        self._check_loaded()
-        return self.k_weights.shape[0]
-
-    def get_eps(self, i, ispin, ik):
-        """Get eigenvalue (in eV) for state i, spin ispin, k-point ik"""
-        return self._eps[ik, ispin, i] * HARTREE_TO_EV
-
-    def get_kvec(self, ik):
-        """Get k-vector for k-point ik
-
-        Returns numpy 3-vector in inverse atomic units
-        """
-        return -self.k_vec[ik, :]
-
-    def get_psi(self, i, ispin, ik):
-        """Get wave function for state i, spin ispin, k-point ik
-
-        Wave function is returned in real space on the (Nx,Ny,Nz) mesh
-        """
-        return np.fft.ifftn(self.get_psi_rec(i, ispin, ik))
-
-    def kweight(self, ik):
-        """Get integration weight for k-point ik"""
-        return self.k_weights[ik]
-
-    @property
-    def cell(self):
-        """Get simulation cell (in bohr units)"""
-        return np.asarray(self.nc_wf["cell"])
-
-
 class extra_waves:
     def __init__(self, inputDict, reader=None, pot=None):
         self.inputDict = inputDict
         if reader is None:
-            self.dft_wv = sx_waves_reader(inputDict, fname="waves.sxb")
+            self.dft_wv = sx_nc_waves_reader(inputDict['working_directory'] + "/waves.sxb")
```

Comment:
It would be better to use pathlib module.

```python
from pathlib import Path
...
self.dft_wv = sx_nc_waves_reader(Path(inputDict['working_directory'])/'waves.sxb')
```